### PR TITLE
Switch to poise

### LINKED ISSUE #17: Command definitions
The current way commands are defined is to say the least a bit silly, having to define each command in two places is double the work.

I've looked into some alternatives and found a total of three ways:

1. The current double defining way
2. The [macros](https://github.com/serenity-rs/serenity/blob/current/examples/e05_command_framework/src/main.rs) defined in `serenity::framework::standard::macros`
3. [Poise](https://github.com/ev3nvy/poise/tree/current/examples/feature_showcase) a command framework for Serenity

### PR DESCRIPTION
**Pull request by samuelhafsteins**
Switch to poise for registering commands
If we choose to use this then this closes #17

### CODE REVIEW

**Comment by DagurB** [comment]
> commit 39e544b9230b8c8a728865d81294ec2de4a0a069 (HEAD -> poise-setup, origin/poise-setup)
> Author: Your Name <samuelhafsteins@gmail.com>
> Date:   Sun Nov 19 21:08:22 2023 +0000
>    remove unused pub

please consider changing your name from "Your Name" to your actual name

**Comment by SvartaHjarta** [comment]
This looks cleaner, it's missing the ephemeral option on replies but I'm in favor of changing to Poise.

File: src/commands/malid.rs [hunk 1]
```diff
@@ -37,28 +35,26 @@ fn format_specific_day(menu: &DailyMenu) -> Result<String, Box<dyn Error>> {
...
   let menu: Vec<DailyMenu> = reqwest::get(url).await?.json::<Vec<DailyMenu>>().await?;
 
   Ok(menu)
 }
 
+#[poise::command(slash_command)]
+pub async fn lunch(ctx: Context<'_>) -> Result<(), Error> {
+  let week_menu = this_weeks_menu().await?;
+
+  ctx.say(format_week_menu(week_menu)?).await?;
```
Comment (SvartaHjarta) [inline src/commands/malid.rs:49]:
```suggestion
  let msg = format_week_menu(week_menu)?;
  ctx.send(|b| {
      b.content(msg)
      .ephemeral(true)
    }).await?;
```

Comment (samuelhafsteins) [reply src/commands/malid.rs:49]: Forgot to place that in, placing it back

**Review by SvartaHjarta — APPROVED** [review]
LGTM

**Pull request merged**